File: 010-Carry Counter/NumberofCarries.cpp
#include <iostream>
using namespace std;

int NumberOfCarries(int num1, int num2) {
    int carry = 0;
    int carryCount = 0;

    while (num1 > 0 || num2 > 0) 
    {
        int digit1 = num1 % 10;
        int digit2 = num2 % 10;

        int sum = digit1 + digit2 + carry;

        if (sum >= 10) 
        {
            carry = sum / 10;
            carryCount++;
        } else
         {
            carry = 0;
        }

        num1 /= 10;
        num2 /= 10;
    }

    return carryCount;
}

int main() {
    int num1 = 451;
    int num2 = 349;
    
    cout << "Number of carries: " << NumberOfCarries(num1, num2) << endl;

    return 0;
}

File: 010-Carry Counter/NumberofCarries.py
def number_of_carries(num1, num2):
    carry = 0
    carry_count = 0

    while num1 > 0 or num2 > 0:
        digit1 = num1 % 10
        digit2 = num2 % 10

        total = digit1 + digit2 + carry

        if total >= 10:
            carry = total // 10
            carry_count += 1
        else:
            carry = 0

        num1 //= 10
        num2 //= 10

    return carry_count

# Example Usage
num1 = 451
num2 = 349
print(f"Number of carries: {number_of_carries(num1, num2)}")
